tools/rv_histogram.c

```diff
@@ -0,0 +1,376 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <getopt.h>
+
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+
+#include "elf.h"
+#include "decode.h"
+
+/* 5 are fuse instructions */
+#define RISCV_INSN_AVAIL_SIZE N_RISCV_INSN_LIST - 5 
+
+static bool ascending_order = false;
+static const char *elf_prog = NULL;
+static const char *optstr = "a";
+
+typedef struct {
+    char insn_str[16];
+    size_t freq;
+} rv_insn_hist_t;
+
+#define RISCV_INSN_STATS           \
```

Comment:
Do do this! You can simply use `RISCV_INSN_LIST`.

Reply:
But `RISCV_INSN_LIST` is concatenate the string not stringizing the string

Reply:
Check `src/common.h` for catenation tricks.